### PR TITLE
Convert array/vector to homogentable

### PR DESCRIPTION
**Pull request by minmingzhu**
## What changes were proposed in this pull request?

1. Convert array ,vector to homogentable.
2. add unit test.

## Does this PR also require the following changes?

- CI
- Documentation
- Example

### CODE REVIEW

File: dev/tools/check-gpu-cpu/check-gpu-cpu.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#include <CL/sycl.hpp>
+#include<iostream>
+#include<cstring>
+
+int check_gpu() {
+    std::cout << "Checkout GPU!" << std::endl;
```
Comment (xwu99) [inline dev/tools/check-gpu-cpu/check-gpu-cpu.cpp:6]:
```suggestion
    std::cout << "Listing GPU devices ..." << std::endl;
```

File: dev/tools/check-gpu-cpu/check-gpu-cpu.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    if (!gpu_found)
+        std::cout << "No GPU found!" << std::endl; 
+
+    return gpu_found ? 0 : -1;
+}
+
+int check_cpu() {
+    std::cout << "Checkout CPU!" << std::endl;
```
Comment (xwu99) [inline dev/tools/check-gpu-cpu/check-gpu-cpu.cpp:31]:
```suggestion
    std::cout << "Listing CPU devices ..." << std::endl;
```

File: dev/tools/check-gpu-cpu/Makefile [hunk 3]
```diff
@@ -1,12 +1,14 @@
-CC := clang
-CXX := clang++
+CC := dpcpp
+CXX := dpcpp
 RM := rm -rf
 
 CFLAGS := -g -Wall -Wno-deprecated-declarations -fPIC -std=c++17 -fsycl
 
-OUTPUT = check-gpu
+LIBS := -L/usr/local/lib64 -l:libstdc++.a
```
Comment (xwu99) [inline dev/tools/check-gpu-cpu/Makefile:7]: I don't think it's a good idea to static link to libstdc++. This lib is a system lib, should be linked automatically.

File: dev/tools/check-gpu-cpu/Makefile [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
 
-OBJS := check-gpu.o
+OUTPUT = check-gpu-cpu
```
Comment (xwu99) [inline dev/tools/check-gpu-cpu/Makefile:9]: better to rename check-gpu-cpu to list-compute-devices

File: dev/ci/ci-local-dpc-test.sh [hunk 5]
```diff
@@ -7,17 +7,23 @@ set -e
...
+
 # Prepare lib resources
 cd $GITHUB_WORKSPACE/mllib-dal
 ../dev/prepare-build-deps-gpu.sh
-./build.sh -p CPU_GPU_PROFILE -q
+./build.sh -p CPU_GPU_PROFILE -q -t DEFAULT_HOST_POLICY
 
 unset LD_LIBRARY_PATH
-./test.sh -p CPU_GPU_PROFILE -q HomogenTableSuite java
+./test.sh -p CPU_GPU_PROFILE -q
```
Comment (xwu99) [inline dev/ci/ci-local-dpc-test.sh:29]: I would not add DEFAULT_HOST_POLICY to build. We will build code for both cpu and gpu devices. 
It should be added to test instead to allow runtime select device. 
I would rename to just "HOST_DEVICE, CPU_DEVICE, GPU_DEVICE", use "-d" for "device".
eg. ./test.sh -p CPU_GPU_PROFILE -d GPU_DEVICE   means run the test for gpu device

File: dev/install-level-zero-ubuntu.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+echo "Installing level-zero components ..."
+sudo apt-get install -y gpg-agent wget
+wget -qO - https://repositories.intel.com/graphics/intel-graphics.key |
+sudo apt-key add -
+ sudo apt-add-repository \
+'deb [arch=amd64] https://repositories.intel.com/graphics/ubuntu focal main'
+sudo apt-get install -y \
+intel-opencl-icd \
+intel-level-zero-gpu level-zero \
+intel-media-va-driver-non-free libmfx1
```
Comment (xwu99) [inline dev/install-level-zero-ubuntu.sh:11]: intel-media-va-driver-non-free libmfx1 those two packages are related to media driver, why install them?

**Review by xwu99 — CHANGES_REQUESTED** [review]

**Review by xwu99 — CHANGES_REQUESTED** [review]

File: mllib-dal/src/main/java/com/intel/oneapi/dal/table/HomogenTable.java [hunk 7]
```diff
@@ -11,78 +11,143 @@ public HomogenTable(long cTable){
...
+        Common.ComputeDevice device = Common.ComputeDevice.get(deviceIndex);
+        impl = new HomogenTableImpl(rowCount, colCount, data,
+                dataLayout, device);
     }
 
     public HomogenTable(long rowCount,
                         long colCount,
                         double[] data,
-                        Common.DataLayout dataLayout){
+                        int layoutIndex,
```
[the remark this answers is not in the document]
Comment (minmingzhu) [reply mllib-dal/src/main/java/com/intel/oneapi/dal/table/HomogenTable.java:99]: [#4](https://github.com/minmingzhu/oap-mllib/pull/4) PR  has used enum. Thus,  do nothing here.

File: mllib-dal/src/main/java/com/intel/oneapi/dal/table/HomogenTable.java [hunk 8]
```diff
@@ -11,78 +11,143 @@ public HomogenTable(long cTable){
     public HomogenTable(long rowCount,
                         long colCount,
                         int[] data,
-                        Common.DataLayout dataLayout){
+                        int deviceIndex){
```
[the remark this answers is not in the document]
Comment (minmingzhu) [reply mllib-dal/src/main/java/com/intel/oneapi/dal/table/HomogenTable.java:14]: https://github.com/minmingzhu/oap-mllib/pull/4 PR has used enum. Thus, do nothing here.

[the remark this answers is not in the document]
Comment (minmingzhu) [reply mllib-dal/src/main/java/com/intel/oneapi/dal/table/HomogenTable.java:14]: https://github.com/minmingzhu/oap-mllib/pull/4 PR has used enum. Thus, do nothing here.

**Pull request merged**